packages/ingest/granule.js

```diff
@@ -623,19 +622,18 @@ async function postS3Object(destination, options) {
 }
 
 /**
- * updates cmrFile metadata for any files being moved
+ * updates cmr xml file with updated file urls
  *
- * @param {Object} cmrFile - cmrFile to be updated
- * @param {Object[]} sourceFiles - array of file objects
- * @param {Object[]} destinations - array of objects defining the destination of granule files
+ * @param {string} granuleId - granuleId
+ * @param {Object} cmrFile - cmr xml file to be updated
+ * @param {Object[]} files - array of file objects
  * @param {string} distEndpoint - distribution enpoint from config
+ * @param {boolean} published - indicate if publish is needed
  * @returns {Promise} returns promise to upload updated cmr file
  */
-async function updateMetadata(cmrFile, sourceFiles, destinations, distEndpoint) {
+async function updateMetadata(granuleId, cmrFile, files, distEndpoint, published) {
```

Comment:
I think this function is getting pretty bulky and maybe we can split it up into a few functions here. i.e. maybe split out posting to CMR

Reply:
Added a new function contructOnlineAccessUrls.